coder_sniffer/Drupal/Sniffs/Arrays/ArraySniff.php

```diff
@@ -109,23 +111,26 @@ public function process(File $phpcsFile, $stackPtr)
         }
 
         if ($isInlineArray === true) {
-            // Check if this array contains at least 3 elements and exceeds the 80
-            // character line length.
-            if ($tokens[$tokens[$stackPtr][$parenthesisCloser]]['column'] > $this->lineLimit) {
+            // Check if this array has more than one element and exceeds the
+            // line length defined by $this->lineLimit.
+            $currentLine = $tokens[$stackPtr]['line'];
+            $tokenCount  = $stackPtr;
+            while ($tokenCount < ($phpcsFile->numTokens - 1) && $tokens[($tokenCount + 1)]['line'] === $currentLine) {
+                $tokenCount++;
+            };
+            $lineLength = ($tokens[$tokenCount]['column'] + $tokens[$tokenCount]['length'] - 1);
+
+            if ($lineLength > $this->lineLimit) {
                 $comma1 = $phpcsFile->findNext(T_COMMA, ($stackPtr + 1), $tokens[$stackPtr][$parenthesisCloser]);
                 if ($comma1 !== false) {
-                    $comma2 = $phpcsFile->findNext(T_COMMA, ($comma1 + 1), $tokens[$stackPtr][$parenthesisCloser]);
-                    if ($comma2 !== false) {
-                        $error = 'If the line declaring an array spans longer than %s characters, each element should be broken into its own line';
-                        $data  = [$this->lineLimit];
-                        $phpcsFile->addError($error, $stackPtr, 'LongLineDeclaration', $data);
-                    }
+                    $error = 'The array declaration line has %s characters (the limit is %s). Each element should be on a new line';
```

Comment:
I think there can be edge cases with arrays with just 1 item like
```php
$array = array(very_long_function_call_i_tell_you($parameter1, $parameter2, $parameter3, $parameter4, $parameter5));
```
But even such a declaration should be split up over multiple lines. Only our error message is a bit inaccurate because the array has only 1 element. We could use "The array declaration line has %s characters (the limit is %s). The array content should be split up over multiple lines"

Reply:
OK, will change the message.

Reply:
I've changed the message text as requested.